blockchain/v2/codec.go

```diff
@@ -0,0 +1,13 @@
+package v2
+
+import (
+	amino "github.com/tendermint/go-amino"
+	"github.com/tendermint/tendermint/types"
+)
+
+var cdc = amino.NewCodec()
+
+func init() {
```

Comment:
thoughts on taking the first steps towards this issue: https://github.com/tendermint/tendermint/issues/2358 of not calling init for registering the types in this PR? @melekes @ebuchman @brapse 

we can start with this reactor, then as we go through other reactors we can do the same.

Reply:
All in for removing inits!